src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -465,6 +426,60 @@ private bool TryRegister(SafeSocketHandle socket, IntPtr handle, out Interop.Err
             return error == Interop.Error.SUCCESS;
         }
 
+        // An on-stack cache of SocketAsyncEngine members needed to process epoll events.
+        // The event handling logic is delegated to a non-inlined processing method
+        // to make sure the runtime doesn't retain internal copies of SocketAsyncContext longer than necessary,
+        // so unreferenced Socket instances always get collected by the GC.
+        // See discussion: https://github.com/dotnet/runtime/issues/37064
+        private struct SocketEventHandler
```

Comment:
Does this have to be in a separate struct? Or can `HandleSocketEvents` be a `NoInlining` method in `SocketAsyncEngine`?

> See discussion: https://github.com/dotnet/runtime/issues/37064

Maybe summarize noteworthy things from the discussion in the comment, and drop the link.

Reply:
I'll try to improve the comments.

A separate struct makes the code cleaner for me (vs. maintaining a bunch of locals and args).  Other, less important aspect: I expect some micro perf. benefits from reduced number of copies when making the call, but I may be wrong on this.